src/playground/sample_data_ingestion/jobs/refresh_job.py

```diff
@@ -0,0 +1,145 @@
+# opensearch-py==2.0.0
+from opensearchpy import OpenSearch
+
+# Standard libraries
+import sys
+from os import environ, listdir, path
+sys.path.append(path.dirname(path.dirname(path.abspath(__file__))))
+from sample_data_indices.sample_data_indices import SampleDataIndex
+from datetime import date, timedelta, datetime
+from shutil import copyfileobj
+from os import remove
+from math import ceil
+from json import load
+import gzip
+
+
+# Array in which unzipped files will be removed
+file_removal_array = []
+
+
+# Utility function for unzipping files
+def unzip(filename):
+  if type(filename) is not str or (type(filename) is str and ".json" not in filename):
+      print("Payload is not a filename or is not a .json file")
+      return None
+  if type(filename) is str and ".gz" not in filename:
+      print("Payload is already unzipped")
+      return filename
+  with gzip.open(filename, 'rt') as fin:
+      with open(filename.split(".gz")[0], 'wb') as fout:
+          copyfileobj(fin, fout)
+  file_removal_array.append(filename.split(".gz")[0])
+  return filename.split(".gz")[0]
+
+
+# Configurations for connection
+HOST = environ.get("HOST")
+PORT = 9200
+AUTH = (environ.get("SAMPLE_DATA_USERNAME"), environ.get("SAMPLE_DATA_PASSWORD"))
+
+
+# Constants
+DIR_PATH = path.dirname(path.realpath(__file__))
+MINUTES_PER_DAY = 1440
+CONFIG_PATH = path.join(DIR_PATH.replace("/jobs", ""), "config")
+
+# Establish connection with OS
+client = OpenSearch(
+    hosts = [{'host': HOST, 'port': PORT}],
+    http_compress = True,
+    http_auth = AUTH,
+    use_ssl = True,
+    verify_certs = False,
+    ssl_assert_hostname = False,
+    ssl_show_warn = False
+)
+
+
+# Opening each config file for each plugin
+for file in listdir(CONFIG_PATH):
+    filename = unzip(file)
+    
+    # If the file is a JSON file
+    if filename:
+        # Open the file
+        with open(path.join(CONFIG_PATH, filename), 'r') as f:
+            config = load(f)
+            # Arguments
+            template = config["template"]
+            mapping = config["mapping"]
+            number = int(config["number"])
+            dataset = []
+            chunk = int(config["chunk"])
+            index_name = config["index_name"]
+            if config["timestamp"] == "None":
+                timestamp = None
+            else:
+                timestamp = config["timestamp"]
+            index_body = config["index_body"]
+            feature = config["feature"]
+            anomaly_percentage = float(config["anomaly_percentage"])
+            avg_min = config["avg_min"]
+            avg_max = config["avg_max"]
+            minutes = int(config["minutes"])
+            abs_min = config["abs_min"]
+            abs_max = config["abs_max"]
```

Comment:
is it poissble to convert json to dataclass, so it can be easily to pass to the function? Thanks

Reply:
I have changed this file to just bundle all `ingest_more()` args into one dict `ingest_args` and by doing this, I can pass key word args as a dict into the function, reducing boilerplate.